core/eval.go

```diff
@@ -860,6 +862,351 @@ func evalQWATCH(args []string, c *Client) []byte {
 	return RESP_OK
 }
 
+// SETBIT key offset value
+func evalSETBIT(args []string) []byte {
+	var err error
+
+	if len(args) != 3 {
+		return Encode(errors.New("ERR wrong number of arguments for 'setbit' command"), false)
+	}
+
+	key := args[0]
+	offset, err := strconv.ParseInt(args[1], 10, 64)
+	if err != nil {
+		return Encode(errors.New("ERR bit offset is not an integer or out of range"), false)
+	}
+
+	value, err := strconv.ParseBool(args[2])
+	if err != nil {
+		return Encode(errors.New("ERR bit is not an integer or out of range"), false)
+	}
+
+	obj := Get(key)
+	requiredByteArraySize := offset/8 + 1
+
+	if obj == nil {
+		obj = NewObj(NewByteArray(int(requiredByteArraySize)), -1, OBJ_TYPE_BYTEARRAY, OBJ_ENCODING_BYTEARRAY)
+		Put(args[0], obj)
+	}
+
+	// handle the case when it is string
+	if assertType(obj.TypeEncoding, OBJ_TYPE_STRING) == nil {
+		return Encode(errors.New("ERR value is not a valid byte array"), false)
+	}
+
+	// handle the case when it is byte array
+	if assertType(obj.TypeEncoding, OBJ_TYPE_BYTEARRAY) == nil {
+		byteArray := obj.Value.(*ByteArray)
+		byteArrayLength := byteArray.Length
+
+		// check whether resize required or not
```

Comment:
Trying to understand this bit...
Are you saying that the offset provided for SET is beyond the size of byte array and you will increase the size to accomodate the bit in a larger byte array?
Is this what Redis does?

Reply:
Yeah I am trying to increase the size to accomodate the bit in a larger byte array?

And regarding redis, they use string that's why they don't have to increase the size or something, but they have included the change of the length of strings in comments: [This](https://github.com/redis/redis/blob/unstable/src/bitops.c#L540)